infrastructure/src/main/java/org/wso2/testgrid/infrastructure/providers/AWSProvider.java

```diff
@@ -206,7 +206,9 @@ private InfrastructureProvisionResult doProvision(InfrastructureConfig infrastru
             return result;
         } catch (WaiterUnrecoverableException e) {
             throw new TestGridInfrastructureException(StringUtil.concatStrings("Error while waiting for stack : "
-                    , stackName, " to complete"), e);
+                    , stackName, " to complete. One or more resources transitioned into a failure/unexpected state.",
+                    "Stacks may contain duplicated resource names. ",
+                    "Please delete redundant resources/Stacks and try again."), e);
```

Comment:
Shouldn't we delete the stack at this moment? when this exception is hit it means stack creation is stuck, otherwise we will have to manually delete the stack. WDYT?

Reply:
Destroying failed infrastructure is common for all infrastructures (AWS, Shell). Hence IMHO,  we can handle this at the TestPlanExecutor when catching the TestGridInfrastructureException which is included in commit d966447. WDYT?

Reply:
+1. And, the current error is not meaningful enough. It's too generic now. Actual AWS cfn creation issues such as VPC creation failed, DB snapshot issue etc. are not logged. 
We should this in another PR.